### PR TITLE
Feature/creation crud methods

### PR DESCRIPTION
**Pull request by luismiguel0529**
## Summary
- Add Crud Book
- Add exception control
- Edit Entity Book
## Trello Card
https://trello.com/c/e89B0v13/18-crear-las-operaciones-crud-de-book

### CODE REVIEW

File: src/main/java/wolox/training/exception/RestExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+@ControllerAdvice
+public class RestExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler({ BookNotFoundException.class })
+    protected ResponseEntity<Object> handleNotFound(Exception e, WebRequest webRequest){
+        return handleExceptionInternal(e,"Book not found",new HttpHeaders(), HttpStatus.NOT_FOUND,webRequest);
+    }
+
+
```
Comment (facucompe) [inline src/main/java/wolox/training/exception/RestExceptionHandler.java]: Extra line jumps
```suggestion
```

File: src/main/java/wolox/training/exception/BookNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package wolox.training.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+//@ResponseStatus(code = HttpStatus.NOT_FOUND,reason = "Book not found")
```
Comment (facucompe) [inline src/main/java/wolox/training/exception/BookNotFoundException.java]: Remove commented code

File: src/main/java/wolox/training/controllers/BookController.java [hunk 3]
```diff
@@ -1,13 +1,63 @@
...
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public Book findById(@PathVariable Long id) throws BookNotFoundException {
+        return bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
+    }
+
+    @PostMapping("/")
```
Comment (facucompe) [inline src/main/java/wolox/training/controllers/BookController.java]: Just leave
```suggestion
    @PostMapping
```

File: src/main/java/wolox/training/controllers/BookController.java [hunk 4]
```diff
@@ -1,13 +1,63 @@
...
 
-@Controller
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
+    @Autowired
+    private BookRepository bookRepository;
+
+    @GetMapping("/")
```
Comment (facucompe) [inline src/main/java/wolox/training/controllers/BookController.java]: Same here:
```suggestion
    @GetMapping
```

**Review by facucompe — CHANGES_REQUESTED** [review]
Great work!!

File: src/main/java/wolox/training/controllers/BookController.java [hunk 5]
```diff
@@ -1,13 +1,63 @@
...
+    @ResponseStatus(HttpStatus.OK)
+    public Iterable findAll(){
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public Book findById(@PathVariable Long id) throws BookNotFoundException {
```
Comment (facucompe) [inline src/main/java/wolox/training/controllers/BookController.java]: I prefer unchecked exceptions. Leave it if you want

**Review by joredudiaz — APPROVED** [review]
Good work! 😄

**Review by facucompe — APPROVED** [review]

**Pull request merged**